tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -501,11 +512,122 @@ class ExecutionTest {
+  TEST_METHOD(LongVector_Initialize_float32);
+
+  // float64 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_float64)
+  TEST_METHOD(LongVector_ScalarMultiply_float64)
+  TEST_METHOD(LongVector_Multiply_float64)
+  TEST_METHOD(LongVector_Add_float64)
+  TEST_METHOD(LongVector_Min_float64)
+  TEST_METHOD(LongVector_Max_float64)
+  // float64 unary ops
+  TEST_METHOD(LongVector_Clamp_float64);
+  TEST_METHOD(LongVector_Initialize_float64);
+
+  // int16 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_int16)
+  TEST_METHOD(LongVector_ScalarMultiply_int16)
+  TEST_METHOD(LongVector_Multiply_int16)
+  TEST_METHOD(LongVector_Add_int16)
+  TEST_METHOD(LongVector_Min_int16)
+  TEST_METHOD(LongVector_Max_int16)
+  // int16 unary ops
+  TEST_METHOD(LongVector_Clamp_int16);
+  TEST_METHOD(LongVector_Initialize_int16);
+
+  // int32 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_int32)
+  TEST_METHOD(LongVector_ScalarMultiply_int32)
+  TEST_METHOD(LongVector_Multiply_int32)
+  TEST_METHOD(LongVector_Add_int32)
+  TEST_METHOD(LongVector_Min_int32)
+  TEST_METHOD(LongVector_Max_int32)
+  // int32 unary ops
+  TEST_METHOD(LongVector_Clamp_int32);
+  TEST_METHOD(LongVector_Initialize_int32);
+
+  // int64 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_int64)
+  TEST_METHOD(LongVector_ScalarMultiply_int64)
+  TEST_METHOD(LongVector_Multiply_int64)
+  TEST_METHOD(LongVector_Add_int64)
+  TEST_METHOD(LongVector_Min_int64)
+  TEST_METHOD(LongVector_Max_int64)
+  // int64 unary ops
+  TEST_METHOD(LongVector_Clamp_int64);
+  TEST_METHOD(LongVector_Initialize_int64);
+
+  // uint16 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_uint16)
+  TEST_METHOD(LongVector_ScalarMultiply_uint16)
+  TEST_METHOD(LongVector_Multiply_uint16)
+  TEST_METHOD(LongVector_Add_uint16)
+  TEST_METHOD(LongVector_Min_uint16)
+  TEST_METHOD(LongVector_Max_uint16)
+  // uint16 unary ops
+  TEST_METHOD(LongVector_Clamp_uint16);
+  TEST_METHOD(LongVector_Initialize_uint16);
+
+  // uint32 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_uint32)
+  TEST_METHOD(LongVector_ScalarMultiply_uint32)
+  TEST_METHOD(LongVector_Multiply_uint32)
+  TEST_METHOD(LongVector_Add_uint32)
+  TEST_METHOD(LongVector_Min_uint32)
+  TEST_METHOD(LongVector_Max_uint32)
+  // uint32 unary ops
+  TEST_METHOD(LongVector_Clamp_uint32);
+  TEST_METHOD(LongVector_Initialize_uint32);
+
+  // uint64 binary ops
+  TEST_METHOD(LongVector_ScalarAdd_uint64)
+  TEST_METHOD(LongVector_ScalarMultiply_uint64)
+  TEST_METHOD(LongVector_Multiply_uint64)
+  TEST_METHOD(LongVector_Add_uint64)
+  TEST_METHOD(LongVector_Min_uint64)
+  TEST_METHOD(LongVector_Max_uint64)
+  // uint64 unary ops
+  TEST_METHOD(LongVector_Clamp_uint64);
+  TEST_METHOD(LongVector_Initialize_uint64);
+
   dxc::DxcDllSupport m_support;
 
   bool m_D3DInitCompleted = false;
   bool m_ExperimentalModeEnabled = false;
   bool m_AgilitySDKEnabled = false;
+  bool m_HLKModeEnabled = false; // Prevent skip logic when running HLK tests.
```

Comment:
There is no differentiation between a TAEF skip or pass in the HLK infra. Skipping when running under the HLK infra is bad. Device capabilities are meant to be queried by the HLK gatherers and tests are only meant to be run if the target device reported support.

Reply:
Sort of a loaded question. I only saw HLK requirements checking for a given shader model when I was poking around. Assuming we have HLK tests that have optional support or support that isn't tied to a shader model, then we might be relying on skip logic in the HLK today. I'd need to audit the test cases we actually register with the HLK to determine if any of them should have more granular selection logic.

Skips probably could be leveraged in the HLK. But the worrisome part is that it would be very easy for a bug in the skip logic to go unnoticed.

That being said...adding this logic for the long vector test cases isn't really signing us up to fix existing tests.

Reply:
You can use `#ifdef _HLK_CONF` to check if code is under HLK build.